src/systems/filecoin_mining/storage_mining/storage_mining_subsystem.go

```diff
@@ -62,21 +60,54 @@ func (sms *StorageMiningSubsystem_I) OnNewRound() {
 }
 
 func (sms *StorageMiningSubsystem_I) tryLeaderElection() {
-	// new election, incremented height
-	Randomness1 := sms.consensus().GetTicketProductionSeed(sms.blockchain().BestChain(), sms.blockchain().LatestEpoch())
-	RandomnessK := sms.consensus().GetElectionProofSeed(sms.blockchain().BestChain(), sms.blockchain().LatestEpoch())
 
+	// Draw randomness from chain for ElectionPoSt and Ticket Generation
+	// Randomness for ticket generation in block production
+	randomness1 := sms.consensus().GetTicketProductionSeed(sms.blockchain().BestChain(), sms.keyStore().OwnerAddress(), sms.blockchain().LatestEpoch())
+
+	// Randomness for ElectionPoSt
+	randomnessK := sms.consensus().GetPoStChallenge(sms.blockchain().BestChain(), sms.keyStore().OwnerAddress(), sms.blockchain().LatestEpoch())
+
+	// TODO: @why @jz align on this
 	for _, worker := range sms.keyStore().Workers() {
-		newTicket := sms.PrepareNewTicket(Randomness1, worker.VRFKeyPair())
-		newEP := sms.DrawElectionProof(RandomnessK, sms.blockchain().LatestEpoch(), worker.VRFKeyPair())
 
-		if sms.consensus().IsWinningElectionProof(newEP, worker.Address()) {
-			sms.blockProducer().GenerateBlock(newEP, newTicket, sms.blockchain().BestChain().HeadTipset(), worker.Address())
+		var input []byte
+		input = append(input, spc.VRFPersonalizationPoSt)
+		input = append(input, randomnessK...)
+
+		postRandomness := worker.VRFKeyPair().Impl().Generate(input).Output()
+		// TODO: add how sectors are actually stored in the SMS
+		allSectors := make([]sector.SectorID, 0)
+
+		challengeTickets := sms.StorageProving().Impl().GeneratePoStCandidates(postRandomness, allSectors)
```

Comment:
we can and probably should get this from `ProvingSet` but we can update this in a separate PR.

Reply:
agreed. if you know how to pull it from the actor, please do!

Reply:
will do this in a separate PR